dhcp_options.go

```diff
@@ -91,26 +91,43 @@ func (odbi *ovndb) dhcpOptionsAddImp(cidr string, options map[string]string, ext
 	return &OvnCommand{operations, odbi, make([][]map[string]interface{}, len(operations))}, nil
 }
 
-func (odbi *ovndb) dhcpOptionsSetImp(cidr string, options map[string]string, external_ids map[string]string) (*OvnCommand, error) {
-
-	row, err := newDHCPRow(cidr, nil, external_ids)
-	if err != nil {
-		return nil, err
+func (odbi *ovndb) dhcpOptionsSetImp(uuid string, options map[string]string, external_ids map[string]string) (*OvnCommand, error) {
+	row := make(OVNRow)
+	dhcpUUID := ""
+	dhcpuuidlist,err := odbi.DHCPOptionsList()
+	if err != nil{
+		return nil, ErrorNotFound
 	}
-
-	dhcpUUID := odbi.getRowUUID(tableDHCPOptions, row)
-	if len(dhcpUUID) == 0 {
+	for _ ,v := range dhcpuuidlist{
+		if v.UUID == uuid{
+			dhcpUUID = uuid
+			break
+		}
+	}
+	if dhcpUUID == ""{
 		return nil, ErrorNotFound
 	}
+	if options != nil {
+		oMap, err := libovsdb.NewOvsMap(options)
+		if err != nil {
+			return nil, err
+		}
+		row["options"] = oMap
+	}
 
-	mutatemap, _ := libovsdb.NewOvsMap(options)
-	mutation := libovsdb.NewMutation("options", opInsert, mutatemap)
+	if external_ids != nil {
+		oMap, err := libovsdb.NewOvsMap(external_ids)
+		if err != nil {
+			return nil, err
+		}
+		row["external_ids"] = oMap
+	}
 	condition := libovsdb.NewCondition("_uuid", "==", libovsdb.UUID{dhcpUUID})
 
 	mutateOp := libovsdb.Operation{
-		Op:        opMutate,
+		Op:        opUpdate,
```

Comment:
The original purpose of this API (I think) is to add option key-value pairs for existing cidr (and external_ids). This patch is changing it to a complete replacement of all options and external_ids. Could you explain the motivation first?

Reply:
if we follow ovn-nbctl commands, we don't eed to replace, but append absent, and change exiting options.

Reply:
got it , wip

Reply:
```
root@esixCell:/home/jacky# ovn-nbctl dhcp-options-set-options ac5a44a0-8c38-4f3b-a861-1e1780d6287a "server_id"="192.168.1.1" "server_mac"="54:54:54:54:54:54" "lease_time"="6000"
root@esixCell:/home/jacky# ovn-nbctl list  DHCP_Options
_uuid               : ac5a44a0-8c38-4f3b-a861-1e1780d6287a
cidr                : "192.168.0.0/24"
external_ids        : {}
options             : {lease_time="6000", server_id="192.168.1.1", server_mac="54:54:54:54:54:54"}
root@esixCell:/home/jacky# ovn-nbctl dhcp-options-set-options ac5a44a0-8c38-4f3b-a861-1e1780d6287a "server_id"="192.168.1.1"  "lease_time"="5000"
root@esixCell:/home/jacky# ovn-nbctl list  DHCP_Options
_uuid               : ac5a44a0-8c38-4f3b-a861-1e1780d6287a
cidr                : "192.168.0.0/24"
external_ids        : {}
options             : {lease_time="5000", server_id="192.168.1.1"}
```
this is ovn-nbctl  dhcp-options-set-options , it just completely replaced . To replace like key-value , we need to contrast change to update, becase options all in one field.

Reply:
Ok if ovn-nbctl replaces options do the same

Reply:
wait , i commit my test env .... I should fix it